DFW.cpp

```diff
@@ -10,298 +10,282 @@
+				packet[i] = Serial1.read();
 
-	while (Serial1.available() > 0) {
-		char clearing = Serial1.read();
+			}
+			if (	packet[0] == 'A' &&
+					packet[1] == 'c' &&
+					packet[6] == 'a'&&
+					packet[9] == 'B' &&
+					packet[12] == 'b') {
+				for (int i = 0; i < 2; i++) {
+					byteBu[i] = packet[10 + i];
+				}
+				for (int i = 0; i < 4; i++) {
+					byteAn[i] = packet[2 + i];
+				}
+
+				/*
+				 for (int i = 0; i < packetSize; i++) {
+				 Serial.print(" , ");
+				 Serial.print(" [");
+				 Serial.print(i);
+				 Serial.print("] ");
+				 if(packet[i]>=65 && packet[i]<=122){
+				 Serial.print("(");
+				 Serial.print((char)packet[i]);
+				 Serial.print(")  ");
+				 }else
+				 if(packet[i]=='\n'){
+				 Serial.print("(\\n)");
+				 }else
+				 if(packet[i]=='\r'){
+				 Serial.print("(\\r)");
+				 }else{
+				 Serial.print("  ");
+				 Serial.print(packet[i], HEX);
+				 }
+				 }
+				 */
+			}
+			lastHeartBeatTime = millis();
+		}
 	}
 
 	//Heartbeat need to read once every 1 seconds
-	hbTime = millis() - lastTime;
+	hbTime = millis() - lastHeartBeatTime;
 	if (hbTime > 1000) {
-		// Serial.println("HB tripped");
-		//Serial.println("HB");
+		Serial.print("\r\nHB tripped clearing data. Diff= ");
+		Serial.print(hbTime);
+		Serial.print(" lastTime = ");
+		Serial.print(lastHeartBeatTime);
+		Serial.print(" current Time = ");
+		Serial.print(millis());
 		digitalWrite(debuginpin, 1);
 		for (int i = 0; i < 2; i++) {
 			byteBu[i] = 127;
 		}
-		for (int i = 0; i <= 4; i++) {
+		for (int i = 0; i < 5; i++) {
 			byteAn[i] = 90;
 		}
+
+	}else{
+		long timeSinceFlash = millis() - flashTime;
+		if(timeSinceFlash>1000){
+			flashTime=millis();
+			digitalWrite(debuginpin, 1);
+		}else{
+			digitalWrite(debuginpin,0);
+		}
 	}
+}
 
+DFW::DFW(int debugpin, void (*autonomous)(long), void (*teleop)(long)) //13 is easiest
+		{
+	pinMode(debugpin, OUTPUT);
+	digitalWrite(debugpin, 0);
+	debuginpin = debugpin;
+	myAutonomous = autonomous;
+	myTeleop = teleop;
+	state = powerup;
+	//printing = false;
 }
 
 DFW::DFW(int debugpin)    //13 is easiest
 		{
 	pinMode(debugpin, OUTPUT);
 	digitalWrite(debugpin, 0);
 	debuginpin = debugpin;
+	myAutonomous = Null;
+	myTeleop = Null;
+	state = powerup;
+	//printing = false;
 }
-
-void DFW::begin(long baud, int port)    //serial1 is the only functional input
+CompetitionState DFW::getCompetitionState(void) {
+	return state;
+}
+void DFW::begin()    //serial1 is the only functional input
 		{
+	Serial1.begin(9600);
+	Serial1.setTimeout(250);
 
-	switch (port) {
-	case 0:
-		Serial.begin(baud);
-		break;
-	case 1:
-		Serial1.begin(baud);
-		Serial1.setTimeout(250);
-		break;
-	case 2:
-		Serial2.begin(baud);
-		break;
-	case 3:
-		Serial3.begin(baud);
-		break;
-	default:
-		Serial1.begin(baud);
-		break;
-	}
 }
 
 int DFW::joystickrv(void) // returns joystick right vertical
 		{
 //Serial.println("jrv");
-	unsigned char stickRv = byteAn[1];
+	unsigned char stickRv = byteAn[0];
 	return (stickRv);
 }

```

Comment:
Can this function just return `byteAn[0]` instead of copying it to another piece of memory?